Intersect.Client.Core/Entities/Resource.cs

```diff
@@ -205,21 +292,20 @@ public override bool Update()
         if (mDisposed)
         {
             LatestMap = null;
-
             return false;
         }
 
         if (Descriptor is { IsDeleted: true } deletedDescriptor)
         {
-            _ = ResourceDescriptor.TryGet(deletedDescriptor.Id, out _descriptor);
-            UpdateFromDescriptor(_descriptor);
+            _ = ResourceDescriptor.TryGet(deletedDescriptor.Id, out var descriptor);
+            Descriptor = descriptor;
+            UpdateCurrentState();
```

Comment:
Can we make the setter for `Descriptor` call `UpdateCurrentState()` instead of calling it separately?